xsd_validate/xsd_validate_activity.go

```diff
@@ -0,0 +1,75 @@
+package xsdvalidate
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"go.artefactual.dev/tools/temporal"
+)
+
+const XSDValidateActivityName = "xsd-validate-activity"
+
+type XSDValidateActivity struct{}
+
+func NewXSDValidateActivity() *XSDValidateActivity {
+	return &XSDValidateActivity{}
+}
+
+type XSDValidateActivityParams struct {
+	// XMLFilePath is the path of the file to be validated.
+	XMLFilePath string
+
+	// XSDFilePath is the path of the XSD file to use for validation.
+	XSDFilePath string
+}
+
+type XSDValidateActivityResult struct {
+	Failures string
+}
+
+// Execute checks an XML file using the XSD file provided and returns error output.
+func (a *XSDValidateActivity) Execute(
+	ctx context.Context,
+	params *XSDValidateActivityParams,
+) (*XSDValidateActivityResult, error) {
+	logger := temporal.GetLogger(ctx)
+	logger.V(1).Info("Executing XMLValidateActivity", "XMLFilePath", params.XMLFilePath)
+
+	// Check XML file using XSD.
+	lintErrors, err := checkXml(params.XMLFilePath, params.XSDFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	return &XSDValidateActivityResult{Failures: lintErrors}, nil
```

Comment:
The max. payload of the results can't exceed 4MB, so I'm wondering if the size of `lintErrors` can be problematic when dealing with big XML files. Not sure if that's a concern right now, maybe something that can be addressed some other time,  but it's not uncommon to see >50MB XML files in Archivematica.
Do we have to return the entire output? Should we just return the first part only?

Reply:
The `--noout` option stops `xmlint` from outputting anything but the errors, so the command output's likely not going to exceed 4MB.

Reply:
But the length of the error stream printed by xmllint can alone exceed that limit too.

I've put an example here: https://gist.github.com/sevein/321eef1b0f94e55abe756e98584a796d, where a person can have a list of tweets. The XSD mistakenly describes a tweet as a positiveInteger, but the document is using strings instead like you'd expect. There are around 37k tweets in that XML document. In this scenario, xmllint's error stream uses 5.76MiB:

```
$ xmllint --schema person.xsd person_invalid.xml --noout 2>&1 | pv -b
[...]
person_invalid.xml:37805: element tweet: Schemas validity error : Element 'tweet': 'Hello world!' is not a valid value of the atomic type 'xs:positiveInteger'.
person_invalid.xml:37805: element tweet: Schemas validity error : Element 'tweet': 'Hello world!' is not a valid value of the atomic type 'xs:positiveInteger'.
5.76MiB
```

Perhaps this is not a problem for your use case (I'm not sure what kind of XML documents you're planning to validate), but I mentioned it because I know this is possible, for example, within the METS XML in Archivematica AIPs.

Reply:
Hrm... maybe in that case passing the filename of a blank file, in which to store error output, to the activity would make more sense?

Reply:
That could work if the reader has access to the filesystem. You could also have the activity store the file elsewhere like MinIO. Some people at DataDog solved this issue with a custom SDK data encoder that stores the outputs in a blobstore: https://github.com/DataDog/temporal-large-payload-codec.

Sounds like a problem that we may have to address at some point, but feel free to ignore for now.